contracts/actions/morpho/MorphoAaveV2Payback.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.10;
+
+import "../../interfaces/morpho/IMorpho.sol";
+import "../../interfaces/aaveV2/IAaveProtocolDataProviderV2.sol";
+import "../ActionBase.sol";
+import "../../utils/TokenUtils.sol";
+import "./helpers/MorphoHelper.sol";
+
+/// @title Payback a token to Morpho
+contract MorphoAaveV2Payback is ActionBase, MorphoHelper {
+    using TokenUtils for address;
+
+    /// @param tokenAddr The address of the token to be payed back
+    /// @param amount Amount of tokens to be payed back
+    /// @param from Where are we pulling the payback tokens amount from
+    /// @param onBehalf For what user we are paying back the debt, defaults to proxy
+    struct Params {
+        address tokenAddr;
+        uint256 amount;
+        address from;
+        address onBehalf;
+    }
+
+    function executeAction(
+        bytes memory _callData,
+        bytes32[] memory _subData,
+        uint8[] memory _paramMapping,
+        bytes32[] memory _returnValues
+    ) public payable virtual override returns (bytes32) {
+        Params memory params = parseInputs(_callData);
+        params.tokenAddr = _parseParamAddr(params.tokenAddr, _paramMapping[0], _subData, _returnValues);
+        params.amount = _parseParamUint(params.amount, _paramMapping[1], _subData, _returnValues);
+        params.from = _parseParamAddr(params.from, _paramMapping[2], _subData, _returnValues);
+        params.onBehalf = _parseParamAddr(params.onBehalf, _paramMapping[3], _subData, _returnValues);
+
+        (uint256 amount, bytes memory logData) = _repay(params);
+        emit ActionEvent("MorphoAaveV2Payback", logData);
+        return bytes32(amount);
+    }
+
+    function executeActionDirect(bytes memory _callData) public payable virtual override {
+        Params memory params = parseInputs(_callData);
+        (, bytes memory logData) = _repay(params);
+        logger.logActionDirectEvent("MorphoAaveV2Payback", logData);
+    }
+
+    function actionType() public pure virtual override returns (uint8) {
+        return uint8(ActionType.STANDARD_ACTION);
+    }
+
+    function _repay(Params memory _params) internal returns (uint256, bytes memory) {
+        _params.amount = _params.tokenAddr.pullTokensIfNeeded(_params.from, _params.amount);
```

Comment:
Does not have the same behavior as the rest of the payback actions.
User debt is increasing, frontend will send a slightly bigger value for payback, this would pull that value and approve it, but then the tokens and the approval would be leftover.

Before pulling figure out the whole debt and pull wholeDebtAmount if amount sent is bigger.